python/tank/descriptor/io_descriptor/git.py

```diff
@@ -147,6 +147,14 @@ def _clone_then_execute_git_commands(self, target_path, commands):
             log.debug("Restoring cwd (to '%s')" % cwd)
             os.chdir(cwd)
 
+        # Cleanup the folder of any git related files.
+        shutil.rmtree(os.path.join(target_path, ".git"), ignore_errors=True)
```

Comment:
Ah, right. It makes sense to keep the .git folder around for a config, but it doesn't for a bundle. A bundle is meant to be a single version of a thing, it is not meant for it to be pulled new versions into, as code would be from a different commit than what its parent folder is.